manifests/machine_info.pp

```diff
@@ -0,0 +1,14 @@
+# @api private
+# @summary This class manages systemd's machine-info file (hostnamectl)
+# @see https://www.freedesktop.org/software/systemd/man/machine-info.html
+class systemd::machine_info {
+  assert_private()
+
+  $systemd::machine_info_settings.each |$option, $value| {
+    file_line { "machine-info ${option}":
```

Comment:
It can get quite tricky to prevent duplicate lines with file_line. I suggest using https://forge.puppet.com/modules/herculesteam/augeasproviders_shellvar

Reply:
I was concerned about adding a new dependency to this module, but if you're not worried I'll get that in place.

Reply:
I've reworked with shellvars and pushed up.  If that looks semi-viable, I can drop the `file_line` implementation.